modules/component/spec/let/let.directive.spec.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  ChangeDetectorRef,
+  Component,
+  NgZone as OriginalNgZone,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+import { CoalescingConfig } from '../../src/core';
+import { EMPTY, NEVER, Observable, of, throwError } from 'rxjs';
+import { async, TestBed } from '@angular/core/testing';
+import { LetDirective } from '../../src/let';
+
+let letDirective: any;
+
+let id = 0;
+
+function MockRequestAnimationFrame(cb: Function) {
+  cb && cb();
+  return ++id;
+}
+
+class NgZone extends OriginalNgZone {
+  constructor() {
+    super({ enableLongStackTrace: false });
+  }
+}
+
+class NoopNgZone {
+  constructor() {
+    // super({enableLongStackTrace: false});
+  }
+}
+
+class MockChangeDetectorRef {
+  public markForCheck(): string {
+    return 'markForCheck';
+  }
+
+  public detectChanges(): string {
+    return 'detectChanges';
+  }
+}
+
+@Component({
+  template: `
+    <ng-container
+      *ngrxLet="value$ as value; $error as error; $complete as complete"
+      >{{ (value | json) || 'undefined' }}</ng-container
+    >
+  `,
+})
+class LetDirectiveTestComponent {
+  cfg: CoalescingConfig = { optimized: false };
+  value$: Observable<number> = of(42);
+}
+
+@Component({
+  template: `
+    <ng-container *ngrxLet="value$; $error as error">{{
+      error.message
+    }}</ng-container>
+  `,
+})
+class LetDirectiveTestErrorComponent {
+  cfg: CoalescingConfig = { optimized: false };
+  value$: Observable<number> = of(42);
+}
+
+@Component({
+  template: `
+    <ng-container *ngrxLet="value$; $complete as complete">{{
+      complete
+    }}</ng-container>
+  `,
+})
+class LetDirectiveTestCompleteComponent {
+  cfg: CoalescingConfig = { optimized: false };
+  value$: Observable<number> = of(42);
+}
+
+let fixtureLetDirectiveTestComponent: any;
+let letDirectiveTestComponent: {
+  cfg: CoalescingConfig;
+  value$: Observable<any> | undefined | null;
+};
+let componentNativeElement: any;
+
+describe('LetDirective', () => {
+  describe('when nexting values', () => {
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
```

Comment:
The setup could be wrapped in a helper function, but not necessary to change these

Reply:
Done https://github.com/ngrx/platform/pull/2046/commits/e101edd38b51213d5b334608517940b941a2ccda